### PR TITLE
ZOB-114 - Fetch user data from entities and entity pairs collections

### PR DESCRIPTION
**Pull request by oleksandrbalan-etn**
* JIRA: https://etnetera.atlassian.net/browse/ZOB-114
* This MR refactors how user data are fetched

### CODE REVIEW

File: lib/common/utils/device_utils.dart [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import 'dart:io';
+import 'package:android_id/android_id.dart';
+import 'package:device_info_plus/device_info_plus.dart';
+
+/// Device related utility functions.
+class DeviceUtils {
+
+  /// Retrieves device ID for Android or iOS.
+  static Future<String?> getId() async {
+    if (Platform.isIOS) {
```
Comment (JaroslavNovak9) [inline lib/common/utils/device_utils.dart:10]: Fungovalo by toto? Možná mi to přijde více elegantnější, ale netuším, zda-li to bude rozumně fungovat.
```suggestion
   switch (Platform.operatingSystem) {
      case 'ios':
        return (await DeviceInfoPlugin().iosInfo).identifierForVendor;
      case 'android':
        return await AndroidId().getId();
      default:
        return null;
    }
```

**Comment by JaroslavNovak9** [comment]
Looks alright by me. Just a single minor comment, otherwise great job!

File: lib/common/utils/device_utils.dart (hunk 1 above)
Comment (oleksandrbalan-etn) [reply lib/common/utils/device_utils.dart]: Yeap, it should definitely work 👍

File: lib/services/ioc_container.dart [hunk 2]
```diff
@@ -3,26 +3,26 @@ import 'package:zachranobed/routes/app_router.dart';
...
-import 'package:zachranobed/services/canteen_service.dart';
-import 'package:zachranobed/services/charity_service.dart';
 import 'package:zachranobed/services/delivery_service.dart';
-import 'package:zachranobed/services/fcm_token_service.dart';
+import 'package:zachranobed/services/entity_pairs_service.dart';
+import 'package:zachranobed/services/entity_service.dart';
 import 'package:zachranobed/services/offered_food_service.dart';
 import 'package:zachranobed/services/shipping_of_boxes_service.dart';
 
 class IoCContainer {
```
Comment (fabiawrc) [inline lib/services/ioc_container.dart:12]: Detail - I would move to `DI` directory and modify the name to` ioc_dependency_container` to keep the names consistent?

**Review by fabiawrc — COMMENTED** [review]
Nice work 👏 Just one comment.

File: lib/services/ioc_container.dart (hunk 2 above)
Comment (oleksandrbalan-etn) [reply lib/services/ioc_container.dart]: Sure, renamed to `ServicesDependencyContainer` to be more specific and moved to "di" subdirectory 👌

**Review by fabiawrc — APPROVED** [review]
All right 👍

**Pull request merged**